iris_grib/tests/unit/message/test__DataProxy.py

```diff
@@ -43,31 +43,5 @@ def test_bitmap__invalid_indicator(self):
             data_proxy._bitmap(section_6)
 
 
-class Test_emptyfetch(tests.IrisGribTest):
```

Comment:
A difficult question! Maybe side with caution and leave it in? We can remove it later on if it causes problems.
On a side note I think we should ASV (aka performance) testing for this sort of thing. I had started thinking about it last week after https://github.com/SciTools/iris/pull/4141.

Reply:
> if it would make sense to leave this in as a "regression" testcase ?

Ok sorry I didn't read this closely enough.

This test will now fail, as we no longer intercept zero-length slicing.
You _could_ modify it to call iris._lazy_data._as_lazy_data + check that that didn't 'visit' the data, but that's really testing Iris (which we do elsewhere), so not really appropriate here.

So, let's scrap it, as you originally suggested. 👍